src/helpers/fetchUrlsData.ts

```diff
@@ -0,0 +1,24 @@
+function fetchSingleUrlData(url: string): Promise<string> {
+  return fetch(url).then((r) => r.json());
+}
+
+export async function fetchUrlsData(
+  urls: Record<string, string>,
+): Promise<Record<string, string>> {
+  /* Convert URLs into an array of promises of tuples consisting of
+   * on one hand, the key of the URL,
+   * and on the other, the response to the URL itself
+   */
+  const keyAndResponsePromises = Object.entries(urls).map(
+    async ([key, url]): Promise<[string, string]> => [
+      key,
+      await fetchSingleUrlData(url),
+    ],
+  );
+  const keyAndResponses = await Promise.all(keyAndResponsePromises);
```

Comment:
this leaves #92 unresolved

Reply:
either put a comment saying that we know there is a bug
or use allSettled

Reply:
There isn't a bug in my opinion, there's a potential improvement which is already tracked in an issue.

I'd add a TODO linking the issue but you specifically asked that we don't do things like this. "That's what issues are for. This adds clutter to the PR and to the code. PRs should do one thing only. Leaving comments for the future is not cool".

I'm not sure what to do if we want to be consistent.

Reply:
a failing await will throw, and that is picked up by react router
changing it to Promise.all() makes it NOT THROUGH, and hence fail silently.